kghDataManagement/views.py

```diff
@@ -1,3 +1,167 @@
-from django.shortcuts import render
+import csv
+import traceback
+from decimal import Decimal
+
+from django.contrib import messages
+from django.db import transaction, IntegrityError
+from django.shortcuts import render, redirect
+from django.http import HttpResponse
 
 # Create your views here.
+from django.urls import reverse
+from django.utils.encoding import smart_str
+from django.views.generic import TemplateView
+
+# TODO, lock down admin only (both get and post request)
+from itemManagement.models import Item
+from simulation_lab import StringUtils
+
+
+class KghUploadPage(TemplateView):
+    # Keys for dictionary from decode_utf8 saved as variables to avoid misspellings
+    ROW_MATERIAL = "material"
+    ROW_OLD_MATERIAL = "oldMaterialNumbers"
+    ROW_PRICE = "maPrice"
+
+    # Keys for dictionary of changes passed to context
+    CHANGE_TITLE = "title"
+    CHANGE_KGH_ID = "kghId"
+    CHANGE_OLD_KGH_ID = "oldKghId"
+    CHANGE_OLD_PRICE = "oldPrice"
+    CHANGE_NEW_PRICE = "newPrice"
+
+    # [kghId, title, oldKghId, oldPrice, newPrice]
+
+    def get(self, request, *args, **kwargs):
+        return render(request, 'kghDataManagement/kgh_upload_page.html')
+
+    def post(self, request, *args, **kwargs):
+        # Note, if this is slow, we can parallelize this
+        try:
+            kghFile = request.FILES['kghFile'] or None
+
+            if kghFile is None or not kghFile.name.endswith('.csv'):
```

Comment:
We probably shouldn't require a specific extension

Reply:
This is just to confirm that the file is a CSV since we only handle those.

Reply:
The problem is the csv extension is a window-ism, especially if we accept by mime type we can't guarantee extension. Lets just rely on the browsers filtering